vault/secret_test.go

```diff
@@ -1,19 +1,90 @@
 package vault
 
-import "testing"
+import (
+	"testing"
+	"math/rand"
+	"time"
+)
 
-const secretName = "/test/secret"
+const letterBytes = "abcdefghijklmnopqrstuvwxyz"
+
+func init() {
+    rand.Seed(time.Now().UnixNano())
+}
 
-// TestSecret tests Secret
 func TestSecret(t *testing.T) {
-	secret, err := dsv.Secret(secretName)
+	path := makeRandomSecretPath()
+	_, cleanup := createSecret(t, path)
+	defer cleanup()
+
+	secret, err := dsv.Secret(path)
+
+	if err != nil {
+		t.Fatalf("Secret for path=%s: %s", path, err)
+		return
+	}
+
+	if secret.Data == nil {
+		t.Error("secret.Data is nil")
+	}
+}
+
+
+func TestCreateSecret(t *testing.T) {
+	path := makeRandomSecretPath()
+	secret, err := dsv.CreateSecret(
+		path, &SecretCreateRequest{Data: map[string]interface{}{"foo": "bar"}},
+	)
 
 	if err != nil {
-		t.Error("calling secrets.Secret:", err)
+		t.Fatalf("CreateSecret for path=%s: %s", path, err)
 		return
+	} else {
+		defer func() { deleteSecret(t, path) }()
 	}
 
 	if secret.Data == nil {
 		t.Error("secret.Data is nil")
 	}
 }
+
+func TestDeleteSecret(t *testing.T) {
+	path := makeRandomSecretPath()
+	_, _ = createSecret(t, path) // no cleanup required as test tries to delete anyway
+
+	err := dsv.DeleteSecret(path)
+	if err != nil {
+		t.Errorf("DeleteSecret for path=%s: %s", path, err)
+	}
+}
+
+// createSecret creates a secret with given path and returns the created secret as well
+// as a cleanup function which should be deferred to remove the secret from the vault.
+func createSecret(t *testing.T, path string) (s *Secret, cleanup func()) {
+	t.Helper()
+	s, err := dsv.CreateSecret(path, &SecretCreateRequest{Data: map[string]interface{}{
+		"foo": "bar",
+	}})
+	if err != nil {
+		t.Fatal(err)
+	}
+	cleanup = func() { deleteSecret(t, path) }
+	return
+}
+
+// deleteSecret deletes the secret given by path from the vault.
+func deleteSecret(t *testing.T, path string) {
+	t.Helper()
+	if err := dsv.DeleteSecret(path); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// makeRandomSecretPath creates a pseudo-random secret path.
+func makeRandomSecretPath() string {
+	b := make([]byte, 10)
+    for i := range b {
+        b[i] = letterBytes[rand.Intn(len(letterBytes))]
+    }
+	return "/test/"+string(b)
```

Comment:
```suggestion
	return "test:"+string(b)
```

If `test` is the first part of the path, then adjust to as above.
While secrets do support slash, all our documentation and tests try to align to `:` as the seperator for DSV.

Reply:
Good to know, addressed in 9d4041a.